source/slang-capture-replay/slang-session.cpp

```diff
@@ -79,8 +87,20 @@ namespace SlangCapture
         slang::IComponentType**         outCompositeComponentType,
         ISlangBlob**                    outDiagnostics)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
-        SlangResult result = m_actualSession->createCompositeComponentType(componentTypes, componentTypeCount, outCompositeComponentType, outDiagnostics);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+
+        Slang::List<slang::IComponentType*> componentTypeList;
+
+        // get the actual component types from our capture wrappers
+        assert(SLANG_OK == getActualComponentTypes(componentTypes, componentTypeCount, componentTypeList));
+
+        slang::IComponentType* compositeComponentType = nullptr;
+        SlangResult result = m_actualSession->createCompositeComponentType(
+                componentTypeList.getBuffer(), componentTypeCount, &compositeComponentType, outDiagnostics);
```

Comment:
I am not sure if this is intentional or a mistake.
The function call to `getActualComponentTypes()` will be removed in release build where `assert()` will be removed.
The behavior looks like `getActualComponentTypes()` will populate `componentTypeList` and it will be used by `createCompositeComponentType()`.
If it is the case, it may not work as expected.

Reply:
Yes, I'm considering define a release version assert actually.
Because I don't want the program proceed when hitting any of assert.

Reply:
We already have one.

Reply:
Hmmm. I think this needs to be wrapped with `do .. while(0)`
```
#define SLANG_RELEASE_ASSERT(VALUE) if(VALUE) {} else SLANG_ASSERT_FAILURE(#VALUE)
```

Reply:
This is what I mentioned yesterday, it will end up to this function:
SLANG_RETURN_NEVER void handleSignal(SignalType type, char const* message);

and it will throw an exception without any useful message printed out.